source/except.d.ts

```diff
@@ -21,4 +21,12 @@ type FooWithoutA = Except<Foo, 'a' | 'c'>;
 
 @category Utilities
 */
-export type Except<ObjectType, KeysType extends keyof ObjectType> = Pick<ObjectType, Exclude<keyof ObjectType, KeysType>>;
+
+type EqualsTest<T> = <A>() => A extends T ? 1 : 0;
+type Equals<A1, A2> = EqualsTest<A2> extends EqualsTest<A1> ? 1 : 0;
```

Comment:
Should this be a separate type? E.g. `EqualsStrict`?

Reply:
Simplified

Reply:
Looks like this type: https://github.com/sindresorhus/type-fest/blob/70fe5242645071808bc556e8aaa0d2ba9e12991b/source/includes.d.ts#L1-L10

Can you move that one out into `source/internal` and use it for now. We can consider exposing it publicly later on.

Reply:
Any reason you're using `1` and `0` here instead of `true` and `false`?

Reply:
Also link to https://stackoverflow.com/a/68963796/64949 for explanation of the type.